openwisp_notifications/static/openwisp-notifications/js/unsubscribe.js

```diff
@@ -0,0 +1,55 @@
+"use strict";
+if (typeof gettext === "undefined") {
+  var gettext = function (word) {
+    return word;
+  };
+}
+
+function updateSubscription(subscribe) {
+  fetch(window.location.href, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ subscribe: subscribe }),
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      if (data.success) {
+        const toggleBtn = document.getElementById("toggle-btn");
+        const statusMessage = document.getElementById("status-message");
+        const confirmationMsg = document.getElementById("confirmation-msg");
+
+        if (subscribe) {
+          statusMessage.textContent = gettext(
+            "You are currently subscribed to notifications."
+          );
+          toggleBtn.textContent = gettext("Unsubscribe");
+          toggleBtn.setAttribute("data-hasSubscribe", "true");
+        } else {
+          statusMessage.textContent = gettext(
+            "You are currently unsubscribed from notifications."
+          );
+          toggleBtn.textContent = gettext("Subscribe");
+          toggleBtn.setAttribute("data-hasSubscribe", "false");
+        }
+
+        confirmationMsg.textContent = gettext(data.message);
```

Comment:
We are displaying output from the API/view here. This is not optimal. We can make the confirmation message independent of the API response. We can similar messages to `statusMessage` in the HTML and hide/show using CSS.